tst/EnergyPlus/unit/InternalHeatGains.unit.cc

```diff
@@ -441,3 +446,70 @@ TEST_F(EnergyPlusFixture, InternalHeatGains_ElectricEquipITE_BeginEnvironmentRes
+
+	std::string const idf_objects = delimited_string({
+		"Version,8.5;",
+		"Zone,Zone1;",
+	});
+
+	ASSERT_FALSE( process_idf( idf_objects ) );
+	EXPECT_FALSE( has_err_output() );
+
+	bool ErrorsFound( false );
+	HeatBalanceManager::GetZoneData( ErrorsFound );
+	ASSERT_FALSE( ErrorsFound );
+	InternalHeatGains::GetInternalHeatGainsInput();
+
+	// Set up a simple convective gain for each gain type
+	int zoneNum = 1;
+	int numGainTypes = DataHeatBalance::NumZoneIntGainDeviceTypes;
+	Array1D< Real64 > convGains;
+	convGains.allocate( numGainTypes );
+	convGains = 0.0;
+	Real64 totConvGains = 0.0;
+	Real64 expectedTotConvGains = 0.0;
+
+	for (int gainType = 1; gainType <= numGainTypes; ++gainType ) {
+		convGains( gainType ) = 100 * gainType;
+		expectedTotConvGains += convGains( gainType );
+		SetupZoneInternalGain( zoneNum, DataHeatBalance::ccZoneIntGainDeviceTypes( gainType ), "Gain", gainType, convGains( gainType ) );
+	}
+
+	InternalHeatGains::UpdateInternalGainValues();
+
+	// Check total of all convective gains
+	InternalHeatGains::SumAllInternalConvectionGains( zoneNum, totConvGains );
+	EXPECT_EQ( totConvGains, expectedTotConvGains );
+
+	// Check subtotals used in zone component loads
+	DataEnvironment::TotDesDays = 1;
+	DataEnvironment::TotRunDesPersDays = 0;
+	DataSizing::CurOverallSimDay = 1;
+	DataGlobals::HourOfDay = 1;
+	DataGlobals::NumOfTimeStepInHour = 10;
+	DataGlobals::TimeStep = 1;
+	OutputReportTabular::AllocateLoadComponentArrays();
+	int timeStepInDay = ( DataGlobals::HourOfDay - 1) * DataGlobals::NumOfTimeStepInHour + DataGlobals::TimeStep;
+		
+	DataGlobals::CompLoadReportIsReq = true;
+	DataGlobals::isPulseZoneSizing = false;
+	InternalHeatGains::GatherComponentLoadsIntGain();
+	totConvGains = OutputReportTabular::peopleInstantSeq( DataSizing::CurOverallSimDay, timeStepInDay, zoneNum ) + OutputReportTabular::lightInstantSeq( DataSizing::CurOverallSimDay, timeStepInDay, zoneNum ) + OutputReportTabular::equipInstantSeq( DataSizing::CurOverallSimDay, timeStepInDay, zoneNum ) + OutputReportTabular::refrigInstantSeq( DataSizing::CurOverallSimDay, timeStepInDay, zoneNum ) + OutputReportTabular::waterUseInstantSeq( DataSizing::CurOverallSimDay, timeStepInDay, zoneNum ) + OutputReportTabular::hvacLossInstantSeq( DataSizing::CurOverallSimDay, timeStepInDay, zoneNum ) + OutputReportTabular::powerGenInstantSeq( DataSizing::CurOverallSimDay, timeStepInDay, zoneNum );
+
+	// Legitimate gain types excluded from this total
+	expectedTotConvGains -= convGains( DataHeatBalance::IntGainTypeOf_ZoneContaminantSourceAndSinkCarbonDioxide ); // this is only used for CO2
+	expectedTotConvGains -= convGains( DataHeatBalance::IntGainTypeOf_ZoneContaminantSourceAndSinkGenericContam ); // this is only used for generic contaminants
+	expectedTotConvGains -= convGains( DataHeatBalance::IntGainTypeOf_DaylightingDeviceTubular ); // this should be included in the component loads as Fenestration Solar or Fenestration Conduction - Instant?
```

Comment:
@JasonGlazer It doesn't appear that conduction from tubular daylighting devices is included in the zone component loads summary.  It's not in the convective gains part, but maybe it's already included somewhere else?

Reply:
I thought that TDD were treated more like a surface than an internal gain.

Reply:
The diffuser is treated as a surface, but apparently the gain from the tube to transition zone(s) is added as convective heat gain component.
```
SetupZoneInternalGain( TDDPipe( PipeNum ).TZone( TZoneNum ), "DaylightingDevice:Tubular", TDDPipe( PipeNum ).Name, IntGainTypeOf_DaylightingDeviceTubular, TDDPipe( PipeNum ).TZoneHeatGain( TZoneNum ) )`

// Distribute absorbed solar gain in proportion to transition zone length
TDDPipe( PipeNum ).TZoneHeatGain( TZoneNum ) = TotTDDPipeGain * ( TDDPipe( PipeNum ).TZoneLength( TZoneNum ) / TDDPipe( PipeNum ).TotLength );

  DaylightingDevice:Tubular,
    Pipe1,                   !- Name
    Dome1,                   !- Dome Name
    Diffuser1,               !- Diffuser Name
    TDD Pipe,                !- Construction Name
    0.3556,                  !- Diameter {m}
    1.4,                     !- Total Length {m}
    0.28,                    !- Effective Thermal Resistance {m2-K/W}
    Daylit Attic Zone,       !- Transition Zone 1 Name
    1.1;                     !- Transition Zone 1 Length {m}
```